R/testthat-ui.R

```diff
@@ -131,22 +131,26 @@ check_versions_match <- function(dep, system_ver, strip_minor = FALSE) {
     )
     return_from(caller_env(), skipped_mismatch_exp(msg, case))
   }
-
-  if (cases_ver < system_ver) {
+  else if (cases_ver < system_ver) {
     msg <- glue(
       "Failed doppelganger was generated with an older { dep } version.
        Please revalidate cases with vdiffr::validate_cases() or vdiffr::manage_cases()"
     )
     return_from(caller_env(), skipped_mismatch_exp(msg, case))
   }
-
-  if (cases_ver > system_ver) {
+  else if (cases_ver > system_ver) {
     msg <- glue(
       "Failed doppelganger was generated with a newer { dep } version.
        Please install { dep } {cases_ver} on your system"
     )
     return_from(caller_env(), skipped_mismatch_exp(msg, case))
   }
+  else if(cases_ver != system_ver)
```

Comment:
Could you give an example of when that happens? (note a space is missing after `if`)

Reply:
This happened when there was an error extracting the version number from the string, yielding an NA in one of the version fields.   In that case, the comparisons fell through, and the function returned when thre *was* an (apparent) version mismatch, but without handling it.

Adding this case would have caught the error closer to the original issue, making it easier to identify the cause.

Reply:
The specific case that highlighted the problem was when the string passed to `as_version` included "package_name:  " at the beginning.   I included a patch to `as_version` to prevent the specific error, but absence of this if-then case caused the issue appear much later in execution, making it harder to identify the source.

Reply:
Thanks, could you add a testcase for `as_version()` that illustrates and checks the problem please. I still don't understand what kind of input you got and why it is different from expected.

Also the new codepath better be `abort("Internal error: Unexpected dependency version structure")` so it's clear what the check is doing.

Reply:
Ok, I added a file `test_as_version` with the troublesome value that was passed to `as_version` by `vdiffr` available on CRAN (`"Cairo: 1.14.6"`).

I also replaced the message with the abort.